src/intelligence_layer/connectors/document_index/document_index.py

```diff
@@ -36,11 +36,13 @@ class IndexConfiguration(BaseModel):
         chunk_overlap: The maximum number of tokens of overlap between consecutive chunks. Must be
             less than `chunk_size`.
         chunk_size: The maximum size of the chunks in tokens to be used for the index.
+        hybrid_index: If set to "bm25", combine vector search and keyword search (bm25) results.
     """
 
     embedding_type: Literal["symmetric", "asymmetric"]
     chunk_overlap: int = Field(default=0, ge=0)
     chunk_size: int = Field(..., gt=0, le=2046)
+    hybrid_index: Literal["bm25"] | None = None
```

Comment:
are you planning to support more hybrid indices in the future?

Reply:
This is a possibility, yes.

Reply:
We want to be able to add new ones.